src/Core/AdminConsole/Services/Implementations/EventIntegrations/IntegrationConfigurationDetailsCacheService.cs

```diff
@@ -32,12 +31,22 @@ public List<OrganizationIntegrationConfigurationDetails> GetConfigurationDetails
         IntegrationType integrationType,
         EventType eventType)
     {
-        var key = new IntegrationCacheKey(organizationId, integrationType, eventType);
-        return _cache.TryGetValue(key, out var value)
-            ? value
-            : new List<OrganizationIntegrationConfigurationDetails>();
-    }
+        var specificKey = new IntegrationCacheKey(organizationId, integrationType, eventType);
+        var allEventsKey = new IntegrationCacheKey(organizationId, integrationType, null);
```

Comment:
❓ We'd never support both existing right? So wouldn't this be fallback logic when there isn't a result found?